File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int l = 0 ,r = n -1,w,h,mx = 0,area;
        while( l < r){
            w = r - l;
            h = min(height[l],height[r]);
            area = w * h;
            mx = max(mx,area);
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return mx;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0 , r = height.length - 1;
        int area = 0 , mx = 0;
        while(l < r){
            int w = r - l;
            int h = Math.min(height[l],height[r]);
            area = w * h;
            mx = Math.max(area,mx);
            if(height[l] < height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return mx;
    }
}